gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxShellTable.java

```diff
@@ -654,5 +655,146 @@ public KnoxShellTable regex(String regex) {
       }
       return table;
     }
+
+    public KnoxShellTable filter(Predicate p) {
+        KnoxShellTable table = new KnoxShellTable();
+        table.headers.addAll(headers);
+        for (List<String> row : rows) {
+          if (p.test(row.get(index))) {
+            table.row();
+            row.forEach(value -> {
+              table.value(value);
+            });
+          }
+        }
+        return table;
+      }
+
+      public KnoxShellTable greaterThan(String comparator) {
+        Predicate<String> p = (s)->s.compareTo(comparator) > 0;
+
+        return filter(p);
+      }
+
+      public KnoxShellTable lessThan(String comparator) {
+          Predicate<String> p = (s)->s.compareTo(comparator) < 0;
+          return filter(p);
+        }
+
+      public KnoxShellTable lessThan(Integer comparator) {
+          Predicate<String> p = (s)->Integer.valueOf(s).compareTo(comparator) < 0;
+          return filter(p);
+        }
+
+      public KnoxShellTable greaterThan(Integer comparator) {
+          Predicate<String> p = (s)->Integer.valueOf(s).compareTo(comparator) > 0;
+          return filter(p);
+        }
+
+
+      public KnoxShellTable greaterThan(Double comparator) {
+        Predicate<String> p = (s)->Double.valueOf(s).compareTo(comparator) > 0;
+
+        return filter(p);
+      }
+
+      public KnoxShellTable lessThan(Double comparator) {
+          Predicate<String> p = (s)->Double.valueOf(s).compareTo(comparator) < 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable lessThan(Float comparator) {
+          Predicate<String> p = (s)->Float.valueOf(s).compareTo(comparator) < 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable greaterThan(Float comparator) {
+          Predicate<String> p = (s)->Float.valueOf(s).compareTo(comparator) > 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable equalTo(String comparator) {
+          Predicate<String> p = (s)->s.compareTo(comparator) == 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable equalTo(Integer comparator) {
+          Predicate<String> p = (s)->Integer.valueOf(s).compareTo(comparator) == 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable equalTo(Double comparator) {
+          Predicate<String> p = (s)->Double.valueOf(s).compareTo(comparator) == 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable equalTo(Float comparator) {
+          Predicate<String> p = (s)->Float.valueOf(s).compareTo(comparator) == 0;
+
+          return filter(p);
+        }
+
+      public KnoxShellTable equalToIgnoreCase(String comparator) {
+          Predicate<String> p = (s)->s.equalsIgnoreCase(comparator);
+
+          return filter(p);
+        }
+
+      public KnoxShellTable greaterThanOrEquals(String comparator) {
+          Predicate<String> p = (s)-> s.compareTo(comparator) > 0 || s.compareTo(comparator) == 0;
```

Comment:
You might be able to reuse the `greaterThan` and `equals` predicates so you don't have to repeat the logic here. Basically make a new predicate the combination of the other predicates.

Reply:
+1 for `return greaterThan(comparator) || equalTo(comparator)`

Reply:
After spending a while looking at this I can't think of a way to do this that would decrease redundancy given the type-specific methods in use - making calls to greaterThan() and equalTo() would return actual table objects where booleans would be needed. Do you have any suggestions?